packages/web3-eth-accounts/src/index.js

```
@@ -138,6 +145,13 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
             );
         }
 
+        if ((tx.chain && !tx.hardfork) || (tx.hardfork && !tx.chain)){
```

If the property ``common`` is defined should ``chain`` and ``hardfork`` be undefined as the documentation of ethereumjs-tx is saying. We can keep this that way if we use the ``hardfork`` and ``chain`` property for the static constructor function of the ``Common`` object.

---

Could you explain this in more detail? Are you saying there should be an additional check here for the case where someone has filled in *all* of the options, e.g. common, hardfork and chain are all defined?

---

Yes, I would throw an error to inform the developer about passed but not used parameters or we use the passed values for the static constructor function on line 181. If we use these parameters then would I add the related default properties to the Eth, Contract, and Accounts module. (``defaultChain``, ``defaultHardfork``, and ``defaultCommon``)